src/gsy_e/models/strategy/scm/smart_meter.py

```diff
@@ -53,9 +53,12 @@ def get_energy_to_buy_kWh(self, time_slot: DateTime) -> float:
     def decrease_energy_to_sell(
             self, traded_energy_kWh: float, time_slot: DateTime, area: "CoefficientArea"):
         """Decrease traded energy from the state and the strategy parameters."""
-        self.state.decrement_energy_requirement(traded_energy_kWh, time_slot, area.name)
+        self.state.decrement_available_energy(traded_energy_kWh, time_slot, area.name)
 
     def decrease_energy_to_buy(
             self, traded_energy_kWh: float, time_slot: DateTime, area: "CoefficientArea"):
         """Decrease traded energy from the state and the strategy parameters."""
-        self.state.decrement_available_energy(traded_energy_kWh, time_slot, area.name)
+        self._energy_params.decrement_energy_requirement(
+            energy_kWh=traded_energy_kWh,
+            time_slot=time_slot,
+            area_name=area.name)
```

Comment:
The old code was calling the wrong method here too: when buying energy, the required energy should be decreased (not the available energy).